mlrun/utils/helpers.py

```diff
@@ -160,6 +160,23 @@ def verify_field_regex(
     return True
 
 
+def validate_tag_name(tag_name: str, raise_on_failure: bool = True) -> bool:
+    """
+    This function is used to validate a tag name for invalid characters using field regex.
+    if raise_on_failure is set True, throws an MLRunInvalidArgumentError if the tag is invalid,
+    otherwise, it returns False
+    """
+    try:
+        mlrun.utils.helpers.verify_field_regex(
+            "artifact.metadata.tag", tag_name, mlrun.utils.regex.tag_name
```

Comment:
This method is pretty generic, and its name represents this (it's not called `validate_artifact_tag_name`), so I would refrain from using hard-coded `artifact.metadata.tag` here but rather receive the name of the field to output on failure.